fgpyo/sam/tests/test_cigar.py

```diff
@@ -0,0 +1,17 @@
+import pytest
+
+from fgpyo.sam import Cigar
+
+cigar = Cigar.from_cigarstring("1M4D45N37X23I11=")
+
+
+def test_cigar_length_exists() -> None:
+    assert len(cigar) == len(cigar.elements)
+
+
+@pytest.mark.parametrize("index", range(-len(cigar), len(cigar)))
+def test_direct_access(index: int) -> None:
```

Comment:
Could you please add a simple test that indexing out of range fails as expected?